File: LeetCode/l283.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n=nums.size(),t=1;
        vector<int> res(n);
        res[0]=1;       
        for(int i=1;i<n;i++)res[i]=res[i-1]*nums[i-1];
        for(int i=n-1;i>=0;i--)res[i]*=t,t*=nums[i];
        return res;
    }
};

File: LeetCode/l283.cs
public class Solution {
    public void swap(ref int a,ref int b){
        int t=a;
        a=b;
        b=t;
    }
    public void MoveZeroes(int[] nums) {
        int l=nums.Length,i=0,j=0;
        while(i<l&&j<l){
            while(i<l&&nums[i]!=0)i++;
            j=i+1;
            while(j<l&&nums[j]==0)j++;
            if(j>=l)break;
            nums[i]=nums[j];nums[j]=0;
            i++;
        }
    }
}